test/manual/keychain_provider/keychain_provider_test.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/cyberark/secretless-broker/internal/app/secretless/providers"
+	plugin_v1 "github.com/cyberark/secretless-broker/pkg/secretless/plugin/v1"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestKeychainProvider(t *testing.T) {
+	var err error
+	var provider plugin_v1.Provider
+
+	name := "keychain"
+
+	// get the environment variables with the test config
+	service := os.Getenv("SERVICE")
+	account := os.Getenv("ACCOUNT")
+	secret := os.Getenv("SECRET")
+
+	options := plugin_v1.ProviderOptions{
+		Name: name,
+	}
+
+	Convey("Can create the Keychain provider", t, func() {
+		provider, err = providers.ProviderFactories[name](options)
+		So(err, ShouldBeNil)
```

Comment:
In fact, probably doesn't even make sense for this to be a test.  If you can't create the provider just halt with an error during the setup phase.

Reply:
Note - this is in several other tests too.